### PR TITLE
[a11y] Table header for Compact List of Tagged Items

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #.

### Summary of Changes
To be accessible a table must have a header. 
See for examle this checklist: https://dequeuniversity.com/checklists/web/tables

Joomla provides a param show_headings for frontend tables. This removes the table header. 

This PR sets the table header visually-hidden so it is still present for assistive technologies. 

I open some similar PRs to make testing easier. 

### Testing Instructions
You need at least one tag 
Make a menu Item of Compact List of Tagged Items.
Set param table_headings in tab "list layouts" to "hide"
See the view and sourcecode ot the table in frontend 

### Actual result BEFORE applying this Pull Request
There is no table heading visible, also in source code.

### Expected result AFTER applying this Pull Request
There is no table heading visible, but it is in sourcecode with class="visually-hidden"

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: components/com_tags/tmpl/tag/list_items.php [hunk 1]
```diff
@@ -57,26 +57,24 @@ class="inputbox" onchange="document.adminForm.submit();"
             </div>
         <?php else : ?>
             <table class="com-tags-tag-list__category category table table-striped table-bordered table-hover">
-                <?php if ($this->params->get('show_headings')) : ?>
-                    <thead>
-                        <tr>
-                            <th scope="col" id="categorylist_header_title">
-                                <?php echo HTMLHelper::_('grid.sort', 'JGLOBAL_TITLE', 'c.core_title', $listDirn, $listOrder); ?>
+                <thead <?php echo $this->params->get('show_headings', '1') ? '' :  'class="visually-hidden"'; ?>>
```
Comment (brianteeman) [inline components/com_tags/tmpl/tag/list_items.php]: Not sure its that important but this code will result in `<thead >`
But the suggested code will produce `<thead>`
```suggestion
                <thead<?php echo $this->params->get('show_headings', '1') ? '' :  ' class="visually-hidden"'; ?>>
```

Comment (chmst) [reply components/com_tags/tmpl/tag/list_items.php]: thx

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on b7dd1d79717e76b793b46d9555f698b70004406c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40138">issues.joomla.org/tracker/joomla-cms/40138</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b7dd1d79717e76b793b46d9555f698b70004406c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40138">issues.joomla.org/tracker/joomla-cms/40138</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40138">issues.joomla.org/tracker/joomla-cms/40138</a>.</sub>

File: components/com_tags/tmpl/tag/list_items.php [hunk 2]
```diff
@@ -57,26 +57,24 @@ class="inputbox" onchange="document.adminForm.submit();"
             </div>
         <?php else : ?>
             <table class="com-tags-tag-list__category category table table-striped table-bordered table-hover">
-                <?php if ($this->params->get('show_headings')) : ?>
-                    <thead>
-                        <tr>
-                            <th scope="col" id="categorylist_header_title">
-                                <?php echo HTMLHelper::_('grid.sort', 'JGLOBAL_TITLE', 'c.core_title', $listDirn, $listOrder); ?>
+                <thead<?php echo $this->params->get('show_headings', '1') ? '' :  ' class="visually-hidden"'; ?>>
```
Comment (Quy) [inline components/com_tags/tmpl/tag/list_items.php]:
```suggestion
                <thead<?php echo $this->params->get('show_headings', '1') ? '' : ' class="visually-hidden"'; ?>>
```

File: components/com_tags/tmpl/tag/list_items.php [hunk 3]
```diff
@@ -57,26 +57,24 @@ class="inputbox" onchange="document.adminForm.submit();"
...
-                    <thead>
-                        <tr>
-                            <th scope="col" id="categorylist_header_title">
-                                <?php echo HTMLHelper::_('grid.sort', 'JGLOBAL_TITLE', 'c.core_title', $listDirn, $listOrder); ?>
+                <thead<?php echo $this->params->get('show_headings', '1') ? '' : ' class="visually-hidden"'; ?>>
+                    <tr>
+                        <th scope="col" id="categorylist_header_title">
+                            <?php echo HTMLHelper::_('grid.sort', 'JGLOBAL_TITLE', 'c.core_title', $listDirn, $listOrder); ?>
+                        </th>
+                        <?php if ($date = $this->params->get('tag_list_show_date')) : ?>
```
Comment (sandewt) [inline components/com_tags/tmpl/tag/list_items.php:65]: I haven't really dug into the code, but I noticed the following:

`if ($date = $this->params->get('tag_list_show_date'))` ?
or 
`if ($date === $this->params->get('tag_list_show_date'))` ?

Comment (Quy) [reply components/com_tags/tmpl/tag/list_items.php:65]: It is correct as it. It is assigning the value to `$date` and then checking for a value.

Comment (sandewt) [reply components/com_tags/tmpl/tag/list_items.php:65]: 👍

**Comment by obuisard** [comment]
Thank you Christiane @chmst !

**Pull request merged**